File: Problems/Algorithms/746. Min Cost Climbing Stairs/min_cost.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let results = (2..cost.len()).into_iter().fold((cost[0], cost[1]), |acc, i| {
            (acc.1, cost[i]+acc.0.min(acc.1))
        });
        results.0.min(results.1)
    }
}

File: Problems/Algorithms/746. Min Cost Climbing Stairs/min_cost.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        dp = [0] * n
        dp[0], dp[1] = cost[0], cost[1]

        for i in range(2, n):
            dp[i] = min(dp[i-2]+cost[i], dp[i-1]+cost[i])

        return min(dp[n-2], dp[n-1])

